File: Java/src/java/EventBus.java
public class EventBus {

    private static EventBus instance;
    private List<Observer> observers = new ArrayList<>();

    public void attach(Observer observer) {
        observers.add(observer);
    }

    private EventBus() {
        ovbservers = new ArrayList<>();
    }

    private static EventBus getInstance() {
        if Objects.isNull(instance) instance = new EventBus();
        return instance;
    }

    public void Post(EventType eventType, String message) {
        for(Observer observer : observers) {
            observer.notify(eventType, message);
        }
    }
}

File: TypeScript/src/EventBus.ts
import { Observer } from "./Observer";
import { EventType } from "./EventType";

export class EventBus {
    private static instance: EventBus | null = null;
    private readonly observers: Observer[] = [];

    private constructor() {}

    public static getInstance(): EventBus {
        if (this.instance === null) {
            this.instance = new EventBus();
        }
        return this.instance;
    }

    public attach(observer: Observer): void {
        this.observers.push(observer);
    }

    public post(eventType: EventType, message: string): void {
        for (const observer of this.observers) {
            observer.notify(eventType, message);
        }
    }
}
